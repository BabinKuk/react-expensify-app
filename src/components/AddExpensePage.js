import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        //console.log(expense);
        // sync with store
        //this.props.dispatch(addExpense(expense));
        // sync with store THIS WAY!!!!!
        this.props.startAddExpense(expense);
        // redirect page
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        );
    }
}

// abstract away dispatch from the component
const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);