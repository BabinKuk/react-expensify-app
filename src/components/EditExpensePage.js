import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        //console.log('editExpense onSubmit');
        // sync with store
        //this.props.dispatch(editExpense(this.props.expense.id, expense));
        // sync with store THIS WAY!!!!!
        this.props.startEditExpense(this.props.expense.id, expense);
        // redirect page
        this.props.history.push('/');
    };

    onRemove = () => {
        console.log('remove');
        // sync with store
        this.props.startRemoveExpense({ id: this.props.expense.id });
        // redirect page
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <h1>Edit Expense</h1>
                <ExpenseForm 
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button onClick={this.onRemove}>Remove</button>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
    // get expense with given id
    expense: state.expenses.find((expense) => {
        return expense.id === props.match.params.id;
    })
});

// abstract away dispatch from the component
const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);