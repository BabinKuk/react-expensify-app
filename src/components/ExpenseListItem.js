// Export a stateless functional component
// description, amount, createdAt

import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import '../locales/hr';
import numeral from 'numeral';

numeral.locale('hr');

export const ExpenseListItem = ({ dispatch, id, description, amount, createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>
            {numeral(amount / 100).format()} 
             - 
            {moment(createdAt).format('DD.MM.YYYY.')}
        </p>
    </div>
);

export default ExpenseListItem;
