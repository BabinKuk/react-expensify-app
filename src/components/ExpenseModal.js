import React from 'react'; 
import Modal from 'react-modal';

const ExpenseModal = (props) => (
    <Modal
        isOpen={!!props.deleteSelected}
        onRequestClose={props.handleClearSelectedExpense}
        contentLabel="Plese confirm to delete selected expense"
        closeTimeoutMS={300}
        className="modal"
    >
        <h3 className="modal__title">Please confirm to delete selected expense</h3>
        {props.deleteSelected && <p className="modal__body">{props.deleteSelected.description}</p>}
        <button className="button" onClick={props.onRemove}>Delete</button>
    </Modal>
);

export default ExpenseModal;
