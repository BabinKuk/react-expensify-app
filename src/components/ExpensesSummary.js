import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import '../locales/hr';
import { getVisibleExpenses } from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expensesCount, expensesTotal, hiddenExpensesCount }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('0,0.00 $');
    const hiddenWord = hiddenExpensesCount === 1 ? 'expense' : 'expenses';
    
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
                {/* display hidden counter only if >0 */}
                {hiddenExpensesCount > 0 && (
                    <p className="page-header__title">Clear search filters to view <span>{hiddenExpensesCount}</span> additional {hiddenWord}</p>
                )}
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
    const hiddenCount = state.expenses.length - visibleExpenses.length;
    
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses),
        hiddenExpensesCount: hiddenCount
    };
};

export default connect(mapStateToProps)(ExpensesSummary);