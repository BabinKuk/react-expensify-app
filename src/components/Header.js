import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
    <header>
        <h1>Expensify</h1>
        <p><NavLink activeClassName="is-active" to="/dashboard" exact={true}>Dashboard</NavLink></p>
        <p><NavLink activeClassName="is-active" to="/create">Create</NavLink></p>
        <p><NavLink activeClassName="is-active" to="/help">Help</NavLink></p>
        <button onClick={startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
