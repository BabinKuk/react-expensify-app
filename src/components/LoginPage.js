import React from 'react';
import { connect} from 'react-redux';
import { startGoogleLogin, startGithubLogin } from '../actions/auth';

export const LoginPage = ({ startGoogleLogin, startGithubLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify App</h1>
            <p>It's time to get your expenses under control.</p>
            <div className="box-layout__buttons">
                {/*<button onClick={startGoogleLogin} className="button">
                    <img src=""></img>
                </button>
                <button onClick={startGithubLogin} className="button">
                    Login with Git
                </button>*/}
                <img src="./images/google-64x.png" onClick={startGoogleLogin} className="button--login" alt="Login with Google"></img>
                <img src="./images/GitHub-Mark-64px.png" onClick={startGithubLogin} className="button--login" alt="Login with Github"></img>
            </div>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startGithubLogin: () => dispatch(startGithubLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);

