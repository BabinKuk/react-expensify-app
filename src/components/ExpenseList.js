import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import getVisibleExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses found</p>
            ) : (
                props.expenses.map((expense) => {
                    return <ExpenseListItem key={expense.id} {...expense} />
                })
            )
        }
    </div>
);

const mapStateToProps = (state) => {
    // display selected data
    return {
        expenses: getVisibleExpenses(state.expenses, state.filters)
    };
};

// HOC with props from the store
export default connect(mapStateToProps)(ExpenseList);
