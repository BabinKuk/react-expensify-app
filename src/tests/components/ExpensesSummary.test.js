import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Test ExpensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={12345} />);
    expect(wrapper).toMatchSnapshot();
});

test('Test ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={1234567890} />);
    expect(wrapper).toMatchSnapshot();
});