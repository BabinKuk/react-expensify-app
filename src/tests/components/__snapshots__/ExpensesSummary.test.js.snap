// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    12.345.678,90 HRK
  </h1>
</div>
`;

exports[`Test ExpensesSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    123,45 HRK
  </h1>
</div>
`;
